### PR TITLE
Fix references to config file in the docs & UX

### PR DESCRIPTION
**Pull request by svlandeg**
## Description
There were some references to `config.cfg` in the docs & error messages that could be pretty confusing if you deliberatly named your config file differently. Also a few places where the docs or comments were just incorrect.

### Types of change
doc fixes.

## Checklist
- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: website/docs/api/cli.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ $ python -m spacy init fill-config [base_path] [output_file] [--diff]
 | Name                   | Description                                                                                                                                                                          |
 | ---------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
 | `base_path`            | Path to base config to fill, e.g. generated by the [quickstart widget](/usage/training#quickstart). ~~Path (positional)~~                                                            |
-| `output_file`          | Path to output `.cfg` file. If not set, the config is written to stdout so you can pipe it forward to a file. ~~Path (positional)~~                                                  |
+| `output_file`          | Path to output `.cfg` file or "-" to write to stdout so you can pipe it forward to a file. ~~Path (positional)~~                                                                     |
```
Comment (polm) [inline website/docs/api/cli.md:178]: Does this still write to stdout if not set?

Also I would remove "forward" here.

**Review by polm — COMMENTED** [review]
Looks good, just had one question.

File: website/docs/api/cli.md (hunk 1 above)
Comment (svlandeg) [reply website/docs/api/cli.md]: I've clarified that "-" is the default

**Comment by polm** [comment]
Latest commit fails with mypy errors? That doesn't make any sense, I'll re-run those...

**Comment by svlandeg** [comment]
The `mypy` errors should hopefully be addressed by https://github.com/explosion/spaCy/pull/9979, let me update, push and let's see what the CI says.

**Pull request merged**